### PR TITLE
Fix `unhashable-member` crash when `lambda` used as a dict key

### LINKED ISSUE #7453: Crash with .AstroidError when parsing a dict with a lambda as the key
### Bug description

pylint crashes over this code

test.py

```python
a = {
    (lambda x: print(x)): 'foo'
}
```

### Configuration

_No response_

### Command used

```shell
> pylint C:\test.py
```

### Pylint output

```shell
************* Module test
test.py:3:0: C0304: Final newline missing (missing-final-newline)
test.py:1:0: C0114: Missing module docstring (missing-module-docstring)
Exception on node <Dict.dict l.1 at 0x1e448164df0> in file 'C:\test.py'
Traceback (most recent call last):
  File "C:\Python310\site-packages\pylint\utils\ast_walker.py", line 90, in walk
    callback(astroid)
  File "C:\Python310\site-packages\pylint\checkers\typecheck.py", line 2036, in visit_dict
    if not is_hashable(k):
  File "C:\Python310\site-packages\pylint\checkers\utils.py", line 1945, in is_hashable
    hash_fn = next(inferred.igetattr("__hash__"))
AttributeError: 'Lambda' object has no attribute 'igetattr'
test.py:1:0: F0002: test.py: Fatal error while checking 'test.py'. Please open an issue in our bug tracker so we address this [...]. (astroid-error)

-8<--------------------------------------------------------------------------

Traceback (most recent call last):
  File "C:\Python310\site-packages\pylint\lint\pylinter.py", line 782, in _lint_file
    check_astroid_module(module)
  File "C:\Python310\site-packages\pylint\lint\pylinter.py", line 1049, in check_astroid_module
    retval = self._check_astroid_module(
  File "C:\Python310\site-packages\pylint\lint\pylinter.py", line 1099, in _check_astroid_module
    walker.walk(node)
  File "C:\Python310\site-packages\pylint\utils\ast_walker.py", line 93, in walk
    self.walk(child)
  File "C:\Python310\site-packages\pylint\utils\ast_walker.py", line 93, in walk
    self.walk(child)
  File "C:\Python310\site-packages\pylint\utils\ast_walker.py", line 90, in walk
    callback(astroid)
  File "C:\Python310\site-packages\pylint\checkers\typecheck.py", line 2036, in visit_dict
    if not is_hashable(k):
  File "C:\Python310\site-packages\pylint\checkers\utils.py", line 1945, in is_hashable
    hash_fn = next(inferred.igetattr("__hash__"))
AttributeError: 'Lambda' object has no attribute 'igetattr'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\Python310\site-packages\pylint\lint\pylinter.py", line 747, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "C:\Python310\site-packages\pylint\lint\pylinter.py", line 784, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```

### Expected behavior

No crash. :)

### Pylint version

```shell
pylint 2.15.2
astroid 2.12.9
Python 3.10.7 (tags/v3.10.7:6cc6b13, Sep  5 2022, 14:08:36) [MSC v.1933 64 bit (AMD64)]
```

### OS / Environment

Windows 10

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Closes #7453

### CODE REVIEW

File: pylint/checkers/utils.py [hunk 1]
```diff
@@ -38,6 +38,15 @@
     nodes.DictComp,
     nodes.GeneratorExp,
 )
+WITH_IGETATTR = (
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/utils.py:41]: Isn't it possible to use ``has_attr("igetattr"`` and have something generic instead ?

**Review by Pierre-Sassoulas — COMMENTED** [review]
Thank you for the quick fix !

File: pylint/checkers/utils.py (hunk 1 above)
Comment (jacobtylerwalls) [reply pylint/checkers/utils.py:41]: Sure, I just figured this was more explicit.

Comment (Pierre-Sassoulas) [reply pylint/checkers/utils.py:41]: > this was more explicit.

Sure but if another class with ``igetattr`` appear we'd need to remember to add it there, which is unlikely to be caught in review, right ?

Comment (jacobtylerwalls) [reply pylint/checkers/utils.py:41]: That makes sense, will change.

**Review by Pierre-Sassoulas — APPROVED** [review]

File: pylint/checkers/utils.py [hunk 2]
```diff
@@ -1950,6 +1950,8 @@ def is_hashable(node: nodes.NodeNG) -> bool:
         for inferred in node.infer():
             if inferred is astroid.Uninferable:
                 return True
+            if not hasattr(inferred, "igetattr"):
```
Comment (DanielNoord) [inline pylint/checkers/utils.py:1953]: `mypy` will complain about this when we add `py.typed` to `astroid`, but that is a problem to fix then I think.

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**